main.go

```diff
@@ -77,7 +78,8 @@ func (hc *HealthCheckWebService) apiGetIsReplica(w http.ResponseWriter, r *http.
 }
 
 var (
-	defaultConnInfo = "host=localhost database=postgres user=postgres sslmode=disable binary_parameters=yes"
+	cfg, _err       = config.ParseConfig("./examples/config.yml")
```

Comment:
After adding `*Config` as the receiver to `Conninfo` func, I am initializing the config here. I am not doing anything with the error here. I found out that I can't use `if` construct outside a function body. Is it okay to leave it like this? Or should I create a function to handle the potential error value?

Reply:
Let's parse the config in the main func, that way we can pass that one instance of config around. For starters we probably want to make the pg replication data source take the config as an arg:

```
func main() {
  cfg, err := config.ParseConfig("./examples/config.yml")
  if err != nil {
    panic(err)
  }
  ds, err := NewPgReplicationDataSource(cfg)
}
```

That makes it easier to add additional args to these structs as time goes on. It also means you don't need to worry about error handling/ parsing this throughout the code.

Example:

```
func buildConninfo(conninfo map[string]string) string {
	cfg, err := config.ParseConfig("./examples/config.yml")
	if err != nil {
		log.Fatalln("Error parsing config:", err)
	}
	ci := fmt.Sprintf("host=%s port=%s database=%s user=%s sslmode=%s binary_parameters=%s", conninfo["host"], conninfo["port"], cfg.Database, cfg.User, cfg.Sslmode, cfg.BinaryParameters)
	return ci
}
```
can become:

```
func  (ds *pgDataSource) buildConninfo(conninfo map[string]string) string {
  return fmt.Sprintf("host=%s port=%s database=%s user=%s sslmode=%s binary_parameters=%s", conninfo["host"], conninfo["port"], 
    ds.cfg.Database, ds.cfg.User, ds.cfg.Sslmode, ds.cfg.BinaryParameters)
}
```

Reply:
Awesome, I think that looks a lot better. Great advice. Let me know if it you have any other suggestions or nits!